tests/LanguageServerCompletion/Unit/Handler/CompletionHandlerTest.php

```diff
@@ -389,14 +421,36 @@ private function create(
             $registry,
             new SuggestionNameFormatter(true),
             $this->createNameImporter($suggestions, $aliases, $importNameTextEdits),
-            $supportSnippets,
+            $this->createClientCapabilities($supportCompletion, $supportSnippets),
             true
         ))->build();
         $tester->textDocument()->open(self::EXAMPLE_URI, self::EXAMPLE_TEXT);
 
         return $tester;
     }
 
+    private function createClientCapabilities(
+        bool $completion = true,
+        bool $supportSnippets = true,
+    ): ClientCapabilities {
+        $capabilities = new ClientCapabilities();
+        $capabilities->textDocument = new TextDocumentClientCapabilities();
```

Comment:
You can call Client capabilities::fromArray() or similar if it helps to generalize.

Reply:
I know about it but in this case building the array would look similar.